### PR TITLE
C++: Refactor StdContainer.qll.

### PR DESCRIPTION
**Pull request by criemen**

### CODE REVIEW

File: cpp/ql/src/semmle/code/cpp/models/implementations/StdContainer.qll [hunk 1]
```diff
@@ -50,7 +88,13 @@ private class StdSequenceContainerConstructor extends Constructor, TaintFunction
  * The standard container function `data`.
  */
 private class StdSequenceContainerData extends TaintFunction {
-  StdSequenceContainerData() { this.hasQualifiedName("std", ["array", "vector"], "data") }
+  StdSequenceContainerData() {
```
Comment (MathiasVP) [inline cpp/ql/src/semmle/code/cpp/models/implementations/StdContainer.qll:91]: I think you want to use the `this.getClassAndName` trick to get a good join-order similar to what we do [here](https://github.com/github/codeql/blob/main/cpp/ql/src/semmle/code/cpp/models/implementations/StdString.qll#L72).

File: cpp/ql/src/semmle/code/cpp/models/implementations/StdContainer.qll [hunk 2]
```diff
@@ -69,8 +110,10 @@ private class StdSequenceContainerData extends TaintFunction {
  */
 private class StdSequenceContainerPush extends TaintFunction {
   StdSequenceContainerPush() {
-    this.hasQualifiedName("std", ["vector", "deque", "list"], "push_back") or
-    this.hasQualifiedName("std", ["deque", "list", "forward_list"], "push_front")
+    this.getClassAndName("push_back") instanceof Array or
```
Comment (geoffw0) [inline cpp/ql/src/semmle/code/cpp/models/implementations/StdContainer.qll:113]:
```suggestion
    this.getClassAndName("push_back") instanceof Vector or
```

File: cpp/ql/src/semmle/code/cpp/models/implementations/StdContainer.qll [hunk 3]
```diff
@@ -69,8 +110,10 @@ private class StdSequenceContainerData extends TaintFunction {
...
+    this.getClassAndName(["push_back", "push_front"]) instanceof Deque or
+    this.getClassAndName("push_front") instanceof ForwardList or
+    this.getClassAndName("push_back") instanceof List
```
Comment (geoffw0) [inline cpp/ql/src/semmle/code/cpp/models/implementations/StdContainer.qll:116]:
```suggestion
    this.getClassAndName(["push_back", "push_front"]) instanceof List
```

**Review by geoffw0 — COMMENTED** [review]
Otherwise LGTM.

**Comment by criemen** [comment]
good catch! I see the tests also found it :)

**Review by geoffw0 — APPROVED** [review]
:+1:

**Pull request merged**